packages/medusa/src/api/admin/products/[id]/route.ts

```diff
@@ -46,6 +57,20 @@ export const POST = async (
     },
   })
 
+  /**
+   * Check 2
+   *
+   * In order to prevent ourselves from reading the `results[0].id`, we
+   * should first check if the workflow returned an array with items
+   * or not.
+   *
+   * If not, should we consider that the product was not found? Or
+   * how do we handle this case.
```

Comment:
but here in theory it would not be possible to be in this scenario right?

Reply:
Not sure I would have this check. `Product not found` is a bit misleading because the product does in fact exist. And if the workflow returns en empty result in this scenario, I would say something is wrong with the workflow. Wdyt?

Reply:
Alright, if we think that as long as the product with id exists, the workflow should return a result, we can remove the second check